qt/widgets/common/src/FunctionTreeView.cpp

```diff
@@ -706,22 +722,116 @@ void FunctionTreeView::addAttributeAndParameterProperties(
+ * @param parentComposite :: If relevant, the composite the function is part of.
+ * @param parentIndex :: If relevant, the index of the function within its
+ * composite.
+ */
+void FunctionTreeView::addParameterTie(
+    QtProperty *property, const IFunction_sptr &function,
+    const std::string &parameterName, const std::size_t &parameterIndex,
+    const CompositeFunction_sptr &parentComposite,
+    const std::size_t &parentIndex) {
+  if (const auto tie = function->getTie(parameterIndex)) {
+    addTieProperty(property, QString::fromStdString(tie->asString()));
+  } else {
+    auto tieAdded = false;
+    if (parentComposite)
+      tieAdded = addParameterTieInComposite(property, parameterName,
+                                            parentComposite, parentIndex);
+
+    if (!tieAdded)
+      addTieProperty(property,
+                     QString::number(function->getParameter(parameterIndex)));
+  }
+}
+
+/**
+ * Add a tie to a function property. Used if a tie is stored within the wider
+ * composite function.
+ * @param property :: A function property.
+ * @param parameterName :: The name of the parameter to be tied.
+ * @param composite :: The composite function containing the tie.
+ * @param index :: The index of the function within the composite function.
+ * @returns true if a tie was found, and a tie property was added.
+ */
+bool FunctionTreeView::addParameterTieInComposite(
+    QtProperty *property, const std::string &parameterName,
+    const CompositeFunction_sptr &composite, const std::size_t &index) {
+  for (auto i = 0u; i < composite->nParams(); ++i) {
+    const auto fullName = "f" + std::to_string(index) + "." + parameterName;
+    if (fullName == composite->parameterName(i)) {
+      if (const auto tie = composite->getTie(i)) {
+        const auto tieStr = QString::fromStdString(tie->asString());
+        addTieProperty(property, tieStr.mid(tieStr.indexOf('=') + 1));
+        return true;
+      }
+      return false;
+    }
+  }
+  return false;
+}
+
+/**
+ * Add a global tie to a function property if one exists for the specified
+ * parameter.
+ * @param property :: A function property.
+ * @param parameterName :: The name of the parameter to check for a global tie.
+ * @param parentComposite :: The composite function the parameter is in. This is
+ * a nullptr if the parameter is not in a composite function.
+ * @param parentIndex :: The index of the parameter within its composite.
+ */
+void FunctionTreeView::addGlobalParameterTie(
+    QtProperty *property, const std::string &parameterName,
+    const CompositeFunction_sptr &parentComposite,
+    const std::size_t &parentIndex) {
+  if (m_multiDomainFunctionPrefix.isEmpty() || m_globalTies.empty())
+    return;
+
+  auto const fullName = getFullParameterName(
+      parameterName, parentComposite ? static_cast<int>(parentIndex) : -1);
+
+  for (auto const &globalTie : m_globalTies) {
+    if (fullName == globalTie.m_parameter) {
```

Comment:
if the same parameter name appears in a composite function multiple times, in different functions. Will this tie them too or will it only tie the ones associated with a specific function? In the most extreme case we may have the same function twice in the composite and only want the first one to have a global.

Reply:
Good question. This will only tie a specific parameter in a composite, even if its function appears twice, because it is comparing a full parameter `f0.f0.Height` instead of just the parameter name `Height`.